docs/src/docs/asciidoc/configuration-model.adoc

```diff
@@ -3,22 +3,231 @@
+
+[[configure-custom]]
+== Customizing the configuration
+
+`OAuth2AuthorizationServerConfigurer` provides the ability to fully customize the security configuration for an OAuth2 authorization server.
+It allows you to specify the core components to use - for example, xref:core-model-components.adoc#registered-client-repository[`RegisteredClientRepository`],  xref:core-model-components.adoc#oauth2-authorization-service[`OAuth2AuthorizationService`], xref:core-model-components.adoc#oauth2-token-generator[`OAuth2TokenGenerator`], and others.
+Furthermore, it allows you to customize the request processing logic for the protocol endpoints – for example, xref:protocol-endpoints.adoc#oauth2-authorization-endpoint[authorization endpoint], xref:protocol-endpoints.adoc#oauth2-token-endpoint[token endpoint], xref:protocol-endpoints.adoc#oauth2-token-introspection-endpoint[token introspection endpoint], and others.
+
+`OAuth2AuthorizationServerConfigurer` provides the following configuration options:
+
+[source,java]
+----
+@Bean
+public SecurityFilterChain authorizationServerSecurityFilterChain(HttpSecurity http) throws Exception {
+	OAuth2AuthorizationServerConfigurer<HttpSecurity> authorizationServerConfigurer =
+		new OAuth2AuthorizationServerConfigurer<>();
+	http.apply(authorizationServerConfigurer);
+
+	authorizationServerConfigurer
+		.registeredClientRepository(registeredClientRepository) <1>
+		.authorizationService(authorizationService) <2>
+		.authorizationConsentService(authorizationConsentService)   <3>
+		.providerSettings(providerSettings) <4>
+		.tokenGenerator(tokenGenerator) <5>
+		.clientAuthentication(clientAuthentication -> { })  <6>
+		.authorizationEndpoint(authorizationEndpoint -> { })    <7>
+		.tokenEndpoint(tokenEndpoint -> { })    <8>
+		.tokenIntrospectionEndpoint(tokenIntrospectionEndpoint -> { })  <9>
+		.tokenRevocationEndpoint(tokenRevocationEndpoint -> { })    <10>
+		.oidc(oidc -> oidc
+			.userInfoEndpoint(userInfoEndpoint -> { })  <11>
+			.clientRegistrationEndpoint(clientRegistrationEndpoint -> { })  <12>
+		);
+
+	return http.build();
+}
+----
+<1> `registeredClientRepository()`: The xref:core-model-components.adoc#registered-client-repository[`RegisteredClientRepository`] to use.
+<2> `authorizationService()`: The xref:core-model-components.adoc#oauth2-authorization-service[`OAuth2AuthorizationService`] to use.
+<3> `authorizationConsentService()`: The xref:core-model-components.adoc#oauth2-authorization-consent-service[`OAuth2AuthorizationConsentService`] to use.
+<4> `providerSettings()`: The <<provider-settings, `ProviderSettings`>> to use.
+<5> `tokenGenerator()`: The xref:core-model-components.adoc#oauth2-token-generator[`OAuth2TokenGenerator`] to use.
+<6> `clientAuthentication()`: The configurer for <<oauth2-client-authentication, OAuth2 Client Authentication>>.
```

Comment:
Add `the` after `for`.

Reply:
I'm not sure `for` is needed here? It would read "The configurer for the OAuth2 Client Authentication". In other cases, it does make sense, e.g. The "configurer for the OAuth2 Authorization endpoint."